CI: Automated Deployment Initial Setup

francinelucca opened the pull request:
Closes #14 , closes #18 , closes #19

Relates to #13 

Introduces Deployment script and config files for automated deployment

#### Changelog

**New**

- Deploy Command Script
- Test & Prod Github Actions
- Test & Prod Service Deployment Config Files
- .cfignore files

**Changed**

- scripts/micromanage-scripts/utils.js: Accept ExecOptions as param in Exec function
- scripts/micromanage-scripts/utils.js: Introduces getPackageByName function
- services/logging-service/src/main/index.ts: serve dummy service

jharvey10 committed:
build: scaffolding for deploy logic

jharvey10 committed:
build: update some deploy script logic

francinelucca committed:
ci: deploy - restructure service-config.json,extend script,create action

francinelucca committed:
ci: add install commander to deploy-services script

francinelucca committed:
Update deploy-services.yml

francinelucca committed:
ci: git switch to tag on deploy script

francinelucca committed:
Merge branch 'deploy-scripts' of github.com:carbon-design-system/carbon-platform into deploy-scripts

francinelucca committed:
ci: do not deploy new services, error out on no ibmcloud, actions config

francinelucca committed:
Merge branch 'main' of github.com:carbon-design-system/carbon-platform into deploy-scripts

francinelucca committed:
Merge branch 'main' of github.com:carbon-design-system/carbon-platform into deploy-scripts

francinelucca committed:
Merge branch 'main' of github.com:carbon-design-system/carbon-platform into deploy-scripts

jharvey10 committed:
build: update micromanage deploy

- Mostly just light refactoring
- Removed a couple props from config file that will be set directly on ibmcloud

francinelucca committed:
Merge branch 'deploy-scripts' of github.com:carbon-design-system/carbon-platform into deploy-scripts

francinelucca committed:
Merge branch 'main' of github.com:carbon-design-system/carbon-platform into deploy-scripts

francinelucca committed:
ci: introduces deployment automation files and extends deploy script

mattrosno reviewed (commented):
Looking great! @jdharvey-ibm I see you're a user in Carbon's IBM Cloud account. Need anything from me?

Instead of `Closes #14 , #18 , #19` I believe you need syntax `Closes #14 , closes #18 , closes #19` for the PR to close multiple issues in one go.

Feel free to use draft PRs instead of prefixing PR name with "WIP".

francinelucca commented:
> Instead of `Closes #14 , #18 , #19` I believe you need syntax `Closes #14 , closes #18 , closes #19` for the PR to close multiple issues in one go.
> Feel free to use draft PRs instead of prefixing PR name with "WIP".

Done! Thanks

francinelucca committed:
refactor: server on logging-service instead of logging package

francinelucca committed:
Merge branch 'main' of github.com:carbon-design-system/carbon-platform into deploy-scripts

francinelucca committed:
fix: copyright notice format on deploy script

francinelucca committed:
build: reinstall @carbon-platform dependencies on deploy script

jharvey10 committed:
feat(micromanage): convert deploy to use dotenv

- Assorted documentation updates

francinelucca committed:
Merge branch 'main' of github.com:carbon-design-system/carbon-platform into deploy-scripts

jharvey10 committed:
fix: merge conflicts

jharvey10 committed:
Merge branch 'deploy-scripts' of github.com:carbon-design-system/carbon-platform into deploy-scripts

jdharvey-ibm commented:
@mattrosno Thanks! I think we're good with cloud stuff. We won't deploy anything just yet. Need a little more testing once this is merged in, but we're super close!

jharvey10 committed:
fix(web-app): lock in specific version of local dep

- Micromanage now uses exact versions for local deps for better consistency across deploys
- This will automatically be kept in sync and respected during nightly versions and periodic deploys

jharvey10 committed:
fix(micromanage): add envvars to deploy scripts

jharvey10 committed:
Merge branch 'main' of github.com:carbon-design-system/carbon-platform into deploy-scripts

francinelucca committed:
Merge branch 'deploy-scripts' of github.com:carbon-design-system/carbon-platform into deploy-scripts

francinelucca committed:
build: rename github actions

jharvey10 committed:
Delete .cfignore

jharvey10 committed:
Delete .cfignore

jharvey10 committed:
Delete .cfignore

jharvey10 committed:
fix(micromanage): correct deploy ignores

francinelucca committed:
build: query apps by org and space,remove data from webapp .cfignore

jharvey10 committed:
refactor(micromanage): adjust deploy structure

- add a helper to get url params for curl commands

jharvey10 committed:
Merge branch 'main' of github.com:carbon-design-system/carbon-platform into deploy-scripts
